src/ggml-cpu/simd-mappings.h

```diff
@@ -0,0 +1,884 @@
+#pragma once
+
+#include "ggml-cpu-impl.h"
```

Comment:
Should this be simply:

```suggestion
#include "ggml-impl.h"
```

Apart from the `GGML_FP32_TO_FP16` and `GGML_FP16_TO_FP32` macros, which are defined in `ggml-impl.h`, it does not seem to need anything eles from `ggml-cpu-impl.h`. Though double-check with the CI just in case.

Reply:
I saw a few lines like these in `ggml-cpu-impl.h`, which define `__F16C__` (which is used in `simd-mappings.h`):

```cpp
// __FMA__ and __F16C__ are not defined in MSVC, however they are implied with AVX2/AVX512
#if defined(_MSC_VER) && (defined(__AVX2__) || defined(__AVX512F__))
#ifndef __FMA__
#define __FMA__
#endif
#ifndef __F16C__
#define __F16C__
#endif
#endif
```